src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -198,9 +198,40 @@ private bool GetCanIncremental(string versionName)
             return LastBuildInfo.Versions.Any(v => v.VersionName == versionName);
         }
 
-        private void ExpandDependency(ChangeList changeList, BuildInfo lastBuildInfo, Dictionary<string, ChangeKindWithDependency> changeItems)
+        private void ExpandDependency(DocumentBuildParameters parameter, BuildInfo lastBuildInfo, Dictionary<string, ChangeKindWithDependency> changeItems)
         {
-            // todo ExpandDependency;
+            string versionName = parameter.VersionName;
+            string dependencyFile = lastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == versionName)?.Dependency;
+
+            if (changeItems == null)
+            {
+                changeItems = new Dictionary<string, ChangeKindWithDependency>();
+            }
```

Comment:
move this outside ExpandDependency()? The dict created here can't be passed to outside context.ChangeDict.